client/src/Dataset/Details/Images/Images.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import MenuItem from '@material-ui/core/MenuItem';
+import Pagination from 'material-ui-flat-pagination';
+
+import { Image } from '../../../common/types';
+import ImageCard from './ImageCard';
+
+interface Props {
+    images: Image[];
+    page: number;
+    pagesCount: number;
+    setPage(offset: number): void;
+    deleteImageAction(id: number): void;
+    annotateImageAction(id: number): void;
+    downloadImageAction(id: number): void;
+}
+
+const Images: React.FC<Props> = ({
+    images,
+    page,
+    pagesCount,
+    setPage,
+    deleteImageAction,
+    annotateImageAction,
+    downloadImageAction,
+}) => {
+    if (images.length === 0) {
+        return <Box textAlign="center">No images found in directory.</Box>;
+    }
+
+    return (
+        <Box>
+            <Box textAlign="center" mb={1}>
+                <Pagination
+                    reduced
+                    size="large"
+                    limit={1}
+                    offset={page}
+                    total={pagesCount}
+                    onClick={(e, offset) => setPage(offset)}
```

Comment:
if you're not using argument given by function it's a common practice to replace it with `_` in TypeScript.

`(_, offset)`